django/db/models/expressions.py

```diff
@@ -534,7 +534,7 @@ def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize
             c.source_expressions[pos] = arg.resolve_expression(query, allow_joins, reuse, summarize, for_save)
         return c
 
-    def as_sql(self, compiler, connection, function=None, template=None):
+    def as_sql(self, compiler, connection, function=None, template=None, arg_joiner=None, **extra_context):
```

Comment:
I'm not seeing the value of `extra_context` here. The purpose seems to be to insert values into the `template` but that can already be customized with the `template` parameter, so why obfuscate things with an extra string interpolation? If the documentation could give an example use case, that would be helpful.